topics/dapp-wallet-msg.md

```diff
@@ -0,0 +1,110 @@
+# Messages
+
+## Permission
+
+A `permission_request` message is sent in order to establish a connection:
+1. The relying party sends `permission_request` to the wallet. The message includes:
+    - basic information about the relying party (`appMetadata`),
+    - a list of networks on which the relying party plans to operate (`networks`),
+    - a list of permission scopes it requires (`scopes`).
+2. The wallet receives the message, presents the details of the to-be-established connection to the user and asks the user to select the account that will be paired in response.
+    - If the user approves the request, the wallet sends `permission_response` back to the relying party. The message includes:
+        - basic information about the wallet (`appMetadata`),
+        - a list of networks on which the user has agreed the relying part can operate, should be a subset of the equivalent from `permission_request` (`networks`),
+        - a list of permission scopes the user has agreed the relying party can be granted, should be a subset of the equivalent from `permission_request` (`scopes`)
+        - the account details (princial or public key (*TBD what is required*) and optional subaccount) of the user selected account (`account`)
+    - If the user rejects the request, the wallet sends `error` back to the relying party with appropriate explanation.
+
+```
+{ // Request
+    type: "permission_request"
+    appMetadata: /* dApp */ {
+        name: String
+        icon: Optional<String>
+    }
+    networks: Array<{
```

Comment:
There is no public testnet, but you could set up a local one. I believe it could come useful to have wallets connect to a local replica, so I would refrain from removing this field entirely.

Thanks for bringing up the CAIP-2 specs, I wasn't aware of such a standard 😄 With that in light, I think we can replace `type` with `chainId` as both fields would effectively hold the same kind of information:
```
{
    chainId: string
    name?: string
    rpcUrl?: string
}
```
We still need `rpcUrl` to let the wallet know what host to use to make calls in case it's running locally, and `name` is there only for the UX.

Reply:
Good point about the local testnets. 👍 
I think it makes sense as suggested, thanks!